src/pkg/control/backup.go

```diff
@@ -0,0 +1,96 @@
+package control
+
+import (
+	"github.com/alcionai/corso/src/pkg/extensions"
+)
+
+// DefaultBackupOptions provides a Backup with the default values set.
+func DefaultBackupOptions() Backup {
+	return Backup{
+		FailureHandling: FailAfterRecovery,
+		DeltaPageSize:   500,
+		Parallelism: Parallelism{
+			CollectionBuffer: 4,
+			ItemFetch:        4,
+		},
+	}
+}
+
+// Backup is the set of options used for backup operations. Each set of options
+// is only applied to the backup operation it's passed to. To use the same set
+// of options for multiple backup operations pass the struct to all operations.
+type Backup struct {
+	// DeltaPageSize controls the quantity of items fetched in each page
+	// during multi-page queries, such as graph api delta endpoints.
+	DeltaPageSize        int32                              `json:"deltaPageSize"`
+	FailureHandling      FailurePolicy                      `json:"failureHandling"`
+	ItemExtensionFactory []extensions.CreateItemExtensioner `json:"-"`
+	Parallelism          Parallelism                        `json:"parallelism"`
+	ToggleFeatures       BackupToggles                      `json:"toggleFeatures"`
```

Comment:
I'm not fully on board with the naming/substruct here. These aren't really feature flags, and `Toggles` doesn't really mean anything.  It's too much a grab bag of "options that are arbitrarily nested instead of top level."  Do you mind fixing it up a bit, since you're here?

Looking at the struct, most of the properties are  related to incremental behavior.  A simple change would be to rename it `IncrementalsConfig`.  The struct could remain more or less where it is, but we'd want to move `ExchangeImmutableIDs` out- maybe to an ExchangeConfig struct?  Or maybe just top level- and move `DeltaPageSize` into it.

Reply:
I agree that the grouping is kind of arbitrary, but looking at what's in the current set of toggles I don't feel like I see a grouping that is less arbitrary. The main things in there related to incrementals is just `DisableIncrementals` and `ForceItemDataDownload`. `DisableDelta`, `RunMigrations`, and `UseDeltaTree` are all related to specific services/versions of backups, not really incrementals as a whole. The same logic kind of applies with `ExchangeImmutableIDs`

Reply:
> are all related to specific services/versions of backups

How much of that is due to incomplete support?  All services should probably support `DisableDelta`, `ForceItemDataDownload`, and `RunMigrations` (though that one probably needs a flag for each migration, really).  By and large we just haven't followed up on making sure that our one-off toggles get full support after they've achieved their v0 purpose.

After that, sure, `UseDeltaTree` stands out as being drive/sharepoint/groups specific (ie: 75% of our service coverage).  But it's still very much a "delta/incremental" behavior set.

No strong feelings here.  It's a minimal impact on our daily lives either way.

Reply:
> A different external service may not have that sort of support or may not map neatly onto that concept

I've been wondering if we'll just end up calling the next provider's incremental changelog `Delta`s for ease of communication.

Reply:
after staring at stuff for a bit, I came up with the following based on what's already defined and the suggestions you presented earlier:
```go
type BackupM365Config struct {
  DeltaPageSize int32
  UseExchangeImmutableIDs bool
  DisableDelta bool
}

type BackupFeatureConfig struct {
  UseDeltaTree bool
  RunMigrations bool
}
```

This presents the delta tree update as a sort of "migration" (in this case a feature migration, not a data one) so groups it there. It also groups what to me feels like the M365-specific stuff into a single place. In addition to this we can make a struct specifically for the two incremental backup flags or just leave them in `BackupConfig`

I'm not really happy with the name of the second struct so if we go with this I'd appreciate name suggestions if you have any

Does this seem reasonable? I'm still a bit hesitant to move `DisableDelta` to be with the incremental backup stuff but don't really have strong feelings on the others

Reply:
ah, I see we commented at roughly the same time

> I've been wondering if we'll just end up calling the next provider's incremental changelog Deltas for ease of communication.

We may. I think the big unknown here is if other services will have a way to disable similar features and if it will have the same sort of behavior that M365 does. For us right now disabling deltas in M365 is more a workaround (due to mailbox being full) than an expectation or a code path that is a full "feature" for how backups should run